spyder/app/tests/test_mainwindow.py

```diff
@@ -1860,5 +1860,27 @@ def test_pylint_follows_file(qtbot, tmpdir, main_window):
         assert fname == pylint_plugin.get_filename()
 
 
+def test_report_comms_error(qtbot, main_window):
+    """Test if a comms error is correctly displayed."""
+    CONF.set('main', 'show_internal_errors', True)
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+    # Create a bogus get_cwd
+    with qtbot.waitSignal(shell.executed):
+        shell.execute('def get_cwd(): import foo')
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("get_ipython().kernel.frontend_comm."
+                      "register_call_handler('get_cwd', get_cwd)")
+    with qtbot.waitSignal(shell.executed):
+        shell.execute('ls')
+
+    error_dlg = main_window.console.error_dlg
+    assert error_dlg is not None
+    assert 'Exception in comms call get_cwd' in error_dlg.error_traceback
+    assert 'No module named' in error_dlg.error_traceback
```

Comment:
The errors on Windows are probably caused because you're closing `error_dlg` after it's displayed by this test. So below this line please add another one that closes it or hides it and then the other tests should pass without problems.